### PR TITLE
fix: swallow transaction rejected error

### PR DESCRIPTION
**Pull request by just-toby**
stop throwing `SwapError`s when the user rejects the Transaction. instead, if we detect that the user rejected, just notify the product code via the return value. for now, we swallow the error and just revert the UI so they can try again. but we may want to show a message inline in this situation

https://uniswaplabs.atlassian.net/browse/WEB-2888?atlOrigin=eyJpIjoiYTg1YWFiZDYwMGM2NDc3NWJmOGRiZGE3NGNmNGQyZTQiLCJwIjoiaiJ9

### CODE REVIEW

File: src/components/Swap/SwapActionButton/useOnSubmit.ts [hunk 1]
```diff
@@ -6,16 +6,20 @@ import { displayTxHashAtom, Field } from 'state/swap'
 import { TransactionInfo, TransactionType } from 'state/transactions'
 import { isAnimating } from 'utils/animations'
 
-/** Submits a transaction. */
+/**
+ * Returns a callback to submit a transaction.
+ *
+ * Returns a boolean indicating whether the transaction was submitted.
```
Comment (zzmp) [inline src/components/Swap/SwapActionButton/useOnSubmit.ts:12]: Could you also specify that this will throw (and not just return false) if something besides a user rejection occurs?

File: src/utils/swapErrorToUserReadableMessage.tsx [hunk 2]
```diff
@@ -12,11 +22,7 @@ export function swapErrorToUserReadableMessage(error: any): string {
     }
   }
 
-  let reason: string | undefined
```
Comment (zzmp) [inline src/utils/swapErrorToUserReadableMessage.tsx:15]: Can this be done in a separate PR? It expands the scope of this change.

File: src/utils/swapErrorToUserReadableMessage.tsx [hunk 3]
```diff
@@ -12,11 +22,7 @@ export function swapErrorToUserReadableMessage(error: any): string {
     }
   }
```
Comment (zzmp) [inline src/utils/swapErrorToUserReadableMessage.tsx:23]: Can this initial check (for USER_REJECTED_REQUEST) be removed? Or are you keeping it for the deprecated non-UniversalRouter codepath?

File: src/hooks/useUniversalRouter.ts [hunk 4]
```diff
@@ -56,7 +63,21 @@ export function useUniversalRouterSwapCallback(trade: InterfaceTrade | undefined
       }
       const gasLimit = calculateGasMargin(gasEstimate)
       response = await provider.getSigner().sendTransaction({ ...tx, gasLimit })
-    } catch (swapError: unknown) {
+    } catch (swapError) {
+      const reason = getReason(swapError)
+      if (
```
Comment (zzmp) [inline src/hooks/useUniversalRouter.ts:68]: Can this logic be moved to a helper function to improve readability/maintainability?

File: src/hooks/useUniversalRouter.ts [hunk 5]
```diff
@@ -21,10 +22,16 @@ interface SwapOptions {
...
+ * Returns a callback to submit a transaction to the universal router.
+ *
+ * The callback returns the TransactionResponse if the transaction was submitted,
+ * or undefined if the user rejected the transaction.
+ **/
 export function useUniversalRouterSwapCallback(trade: InterfaceTrade | undefined, options: SwapOptions) {
   const { account, chainId, provider } = useWeb3React()
 
-  return useCallback(async (): Promise<TransactionResponse> => {
+  return useCallback(async (): Promise<TransactionResponse | undefined> => {
```
Comment (zzmp) [inline src/hooks/useUniversalRouter.ts:34]: nit: I think it would be clearer if you returned something concrete, not `undefined`. Could you create a `Symbol('Rejected transaction')` and return that instead? Or even just return `false` or `null`?

**Review by zzmp — COMMENTED** [review]
Comments to improve readability - none of them are actually blocking, code looks good 😄

File: src/components/Swap/SwapActionButton/SwapButton.tsx [hunk 6]
```diff
@@ -90,7 +90,9 @@ export default function SwapButton({ disabled }: { disabled: boolean }) {
       })
 
       // Only close the review modal if the swap submitted (ie no-throw).
-      setOpen(false)
+      if (submitted) {
```
Comment (zzmp) [inline src/components/Swap/SwapActionButton/SwapButton.tsx:93]: I think @infredible has an opinion (for a future design) on what the UX should be for user rejection, it may be good to check in.

File: src/utils/swapErrorToUserReadableMessage.tsx (hunk 3 above)
Comment (just-toby) [reply src/utils/swapErrorToUserReadableMessage.tsx:23]: i think we should keep it so that `swapErrorToUserReadableMessage` is complete - this is the message it should return in this case.

File: src/utils/swapErrorToUserReadableMessage.tsx (hunk 2 above)
Comment (just-toby) [reply src/utils/swapErrorToUserReadableMessage.tsx:15]: we need this change to actually detect user rejection. the change in this file is just a small refactor so i don't duplicate code.

**Review by tinaszheng — APPROVED** [review]
